File: Exercism/java/bob/src/main/java/Bob.java
/**
 * Created by cajetan on 6/3/17.
 */

public class Bob {
    public String hey(String message) {
        if (message.isEmpty() || message.chars().allMatch(Character::isWhitespace))
            return "Fine. Be that way!";

        if (message.chars().noneMatch(Character::isAlphabetic)) {
           if (message.endsWith("?"))
               return "Sure.";
           else
               return "Whatever.";
        }

        if (message.chars()
                .allMatch(c -> !(Character.isAlphabetic(c)) ||
                        Character.isUpperCase(c))) {

            return "Whoa, chill out!";
        }

        if (message.endsWith("?"))
            return "Sure.";

        return "Whatever.";
    }
}

File: Exercism/dart/bob/lib/bob.dart

class Bob {
  String response(String input) {
    input = input.trim();

    if (input.isEmpty)
      return "Fine. Be that way!";

    // No lowercase letters
    if (RegExp(r"([a-z])").allMatches(input).isEmpty) {
      // No uppercase letters
      if (RegExp(r"([A-Z])").allMatches(input).isEmpty) {
        // No letters at all
        if (input.endsWith("?"))
          return "Sure.";
        return "Whatever.";
      }

      // Only uppercase letters
      if (input.endsWith("?"))
        return "Calm down, I know what I\'m doing!";
      return "Whoa, chill out!";
    }

    if (input.endsWith("?"))
      return "Sure.";
    return "Whatever.";
  }
}
